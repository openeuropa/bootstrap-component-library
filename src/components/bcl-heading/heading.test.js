import { merge, renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
const template = "@oe-bcl/bcl-heading/heading.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Heading", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ title: "A title created by the bcl heading template" })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with custom tag`, () => {
    expect.assertions(1);

    return expect(
      render({
        title: "A title created by the bcl heading template",
        title_tag: "h3",
      })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with a link`, () => {
    expect.assertions(1);

    return expect(
      render({
        title_link: {
          path: "/example.html",
          label: "A title created by the bcl heading template",
        },
      })
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with a link and a title`, () => {
    expect.assertions(1);

    return expect(
      render({
        title: "A title created by the bcl heading template",
        title_link: {
          path: "/example.html",
        },
      })
    ).resolves.toMatchSnapshot();
  });
});
