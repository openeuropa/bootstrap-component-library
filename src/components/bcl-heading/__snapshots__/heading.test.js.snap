// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Heading renders correctly 1`] = `
<jest>
  <h2
    class="fw-bold mb-4 bcl-heading"
  >
    A title created by the bcl heading template
  </h2>
</jest>
`;

exports[`OE - Heading renders correctly with a link 1`] = `
<jest>
  <h2
    class="fw-bold mb-4 bcl-heading"
  >
    <a
      class="standalone"
      href="/example.html"
    >
      A title created by the bcl heading template
    </a>
  </h2>
</jest>
`;

exports[`OE - Heading renders correctly with a link and a title 1`] = `
<jest>
  <h2
    class="fw-bold mb-4 bcl-heading"
  >
    <a
      class="standalone"
      href="/example.html"
    >
      A title created by the bcl heading template
    </a>
  </h2>
</jest>
`;

exports[`OE - Heading renders correctly with custom tag 1`] = `
<jest>
  <h3
    class="fw-bold mb-4 bcl-heading"
  >
    A title created by the bcl heading template
  </h3>
</jest>
`;
